nemo/collections/asr/models/slu_models.py

```diff
@@ -0,0 +1,325 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from pathlib import Path
+from typing import Dict, Optional
+from urllib.error import HTTPError
+
+import torch
+from omegaconf import DictConfig, open_dict
+
+import nemo.collections.asr as nemo_asr
+from nemo.collections.asr.data.audio_to_text_dali import DALIOutputs
+from nemo.collections.asr.losses.slu_losses import SmoothedNLLLoss
+from nemo.collections.asr.models.ctc_bpe_models import EncDecCTCModelBPE
+from nemo.collections.asr.parts.utils.slu_utils import SearcherConfig, SequenceGenerator, get_seq_mask
+from nemo.collections.common.parts.adapter_modules import LinearAdapterConfig
+from nemo.core import adapter_mixins
+from nemo.core.classes.common import Serialization, typecheck
+from nemo.core.neural_types import AudioSignal, LabelsType, LengthsType, LogprobsType, NeuralType, SpectrogramType
+from nemo.utils import logging
+
+
+class SLUIntentSlotBPEModel(EncDecCTCModelBPE):
+    def __init__(self, cfg: DictConfig, trainer=None):
+        if hasattr(cfg, "adapter") and getattr(cfg.adapter, "enabled", False):
+            logging.info("Using adapters...")
+            with open_dict(cfg):
+                adapter_metadata = adapter_mixins.get_registered_adapter(cfg.encoder._target_)
+                if adapter_metadata is not None:
+                    cfg.encoder._target_ = adapter_metadata.adapter_class_path
+
+        super().__init__(cfg=cfg, trainer=trainer)
+
+        # Init encoder from pretrained model
+        if self.cfg.pretrained_encoder.model is not None:
+            if Path(self.cfg.pretrained_encoder.model).is_file():
+                logging.info(f"Loading pretrained encoder from local: {self.cfg.pretrained_encoder.model}")
+                pretraind_model = nemo_asr.models.SpeechEncDecSelfSupervisedModel.restore_from(
+                    restore_path=self.cfg.pretrained_encoder.model, map_location=torch.device("cpu")
+                )
+                self.encoder.load_state_dict(pretraind_model.encoder.state_dict(), strict=False)
+                del pretraind_model
+            else:
+                logging.info(f"Loading pretrained encoder from NGC: {self.cfg.pretrained_encoder.model}")
+                try:
+                    pretraind_model = nemo_asr.models.SpeechEncDecSelfSupervisedModel.from_pretrained(
+                        model_name=self.cfg.pretrained_encoder.model, map_location=torch.device("cpu")
+                    )
+                    self.encoder.load_state_dict(pretraind_model.encoder.state_dict(), strict=False)
+                    del pretraind_model
+                except HTTPError:
```

Comment:
Umm dont catch this error. If user wanted to load pretraind model, error out if it cant be done for some reason.